release/src/router/rom/webs_scripts/merlin_webs_update.sh

```diff
@@ -18,7 +18,7 @@ fi
 current_base=$(nvram get firmver | sed "s/\.//g")
 current_firm=$(nvram get buildno | cut -d. -f1)
 current_buildno=$(nvram get buildno | cut -d. -f2)
-current_extendno=$(nvram get extendno | sed "s/-g.*//" | sed "s/_.*//" | sed "s/alpha\|beta/-1/")
+current_extendno=$(nvram get extendno | awk -F'[_-]' '{n=$1} /[aA]lpha|[bB]eta/{n--} END{print n}') #Extract extendno, subtract value by 1 if it contains "alpha/beta", remove all other values such as "_rog" or "-g*"
```

Comment:
This has never properly worked:

# Define the extendnos as a space-separated string
extendnos="2_alpha1-g4889cee45a 0 2_alpha3 3 0_rog 0_alpha1 0_rog_alpha1 1_alpha1 1_rog_alpha1 0_Alpha1 0_Rog_Alpha1 0_Beta2 1_beta3"

# Loop through each extendno
for extendno in $extendnos; do
  current_extendno=$(echo "$extendno" | sed "s/-g.*//" | sed "s/_.*//" | sed "s/alpha\|beta/-1/")
  echo "Original: $extendno -> Processed: $current_extendno"
done


Original: 2_alpha1-g4889cee45a -> Processed: 2
Original: 0 -> Processed: 0
Original: 2_alpha3 -> Processed: 2
Original: 3 -> Processed: 3
Original: 0_rog -> Processed: 0
Original: 0_alpha1 -> Processed: 0
Original: 0_rog_alpha1 -> Processed: 0
Original: 1_alpha1 -> Processed: 1
Original: 1_rog_alpha1 -> Processed: 1
Original: 0_Alpha1 -> Processed: 0
Original: 0_Rog_Alpha1 -> Processed: 0
Original: 0_Beta2 -> Processed: 0
Original: 1_beta3 -> Processed: 1

Reply:
# Proposed solution:

Define the extendnos as a space-separated string
```
extendnos="2_alpha1-g4889cee45a 0 2_alpha3 3 0_rog 0_alpha1 0_rog_alpha1 1_alpha1 1_rog_alpha1 0_Alpha1 0_Rog_Alpha1 0_Beta2 1_beta3"
```

Loop through each extendno
```
for extendno in $extendnos; do
  # Process each extendno in a single line
  processed_number=$(echo "$extendno" | awk -F'[_-]' '{n=$1} /[aA]lpha|[bB]eta/{n--} END{print n}')
  echo "Original: $extendno -> Processed: $processed_number"
done
```

# Output:
Original: 2_alpha1-g4889cee45a -> Processed: 1
Original: 0 -> Processed: 0
Original: 2_alpha3 -> Processed: 1
Original: 3 -> Processed: 3
Original: 0_rog -> Processed: 0
Original: 0_alpha1 -> Processed: -1
Original: 0_rog_alpha1 -> Processed: -1
Original: 1_alpha1 -> Processed: 0
Original: 1_rog_alpha1 -> Processed: 0
Original: 0_Alpha1 -> Processed: -1
Original: 0_Rog_Alpha1 -> Processed: -1
Original: 0_Beta2 -> Processed: -1
Original: 1_beta3 -> Processed: 0

Reply:
@Martinski4GitHub 

Thoughts?

Reply:
> 
> ```
> for extendno in $extendnos; do
>   # Process each extendno in a single line
>   processed_number=$(echo "$extendno" | awk -F'[_-]' '{n=$1} /[aA]lpha|[bB]eta/{n--} END{print n}')
>   echo "Original: $extendno -> Processed: $processed_number"
> done
> ```
>
Yes, another good catch. Your solution works with underscore & hyphen chars as delimiters & whether the "Alpha/Beta" indicator strings start with lower or upper case so it's a very good "universal" fix.

Reply:
> @Martinski4GitHub
> 
> Thoughts?
>
The only things I would change are the following:

1) Enclose the right statement of the variable assignment in double quotes.

 2) Move the comment to a separate line **before** the modified/fixed code. I know this is subjective but the preferred coding standards call for placing comments **before** the fix, especially when the line is already very long (e.g. > 80 chars).

For example:

```
#Extract extendno, subtract value by 1 if it contains "alpha/beta", remove all other values such as "_rog" or "-g*"
current_extendno="$(nvram get extendno | awk -F'[_-]' '{n=$1} /[aA]lpha|[bB]eta/{n--} END{print n}')"
```

Just my 2 cents.

Reply:
@Martinski4GitHub 

Your 2 cents always make sense to me. Or should I say.. always makes "cents" to me 😂

I'll submit the changes.

Reply:
@Martinski4GitHub 

Done in commit: https://github.com/gnuton/asuswrt-merlin.ng/pull/600/commits/9af68e196f2791525e54260b705487b5f5e63263